Delete student from team

whelan-low opened the pull request:
Fixes #102.

Usage: /delete_student_from_team module/MODULE_CODE tutorial/TUTORIAL_CLASS team/TEAM_NAME

Note: changed command name from `/delete_from_team` to `/delete_student_from_team` for consistency reasons

Also refactored some of the common team-based messages into a new class TeamMessages.

TODO:
- [x] Add more test cases
- [x] Fix help formatting

whelan-low committed:
Merge commit 'a7d285ef69be5e16cee7ae1d288a452e0f8638ae'

whelan-low committed:
Merge remote-tracking branch 'origin/master'

whelan-low committed:
Merge branch 'AY2324S2-CS2103T-T09-4:master' into master

whelan-low committed:
Merge branch 'AY2324S2-CS2103T-T09-4:master' into master

whelan-low committed:
Merge branch 'AY2324S2-CS2103T-T09-4:master' into master

whelan-low committed:
Add assertion test

whelan-low committed:
Merge remote-tracking branch 'origin/update-model-uml'

whelan-low committed:
Merge branch 'AY2324S2-CS2103T-T09-4:master' into master

whelan-low committed:
Add delete student from team command

whelan-low committed:
Add more test cases and fix /delete_student_from_team message formatting

whelan-low committed:
Add successful test cases

mahadhir247 reviewed (commented):
LGTM! Good job on delete command 👍 . I did notice there were quite a bit of message refactoring. Perhaps in the future you could create a separate PR for this so as to not make the PR too big.

whelan-low commented on src/main/java/seedu/address/logic/commands/deletestudentfromteamcommands/DeleteStudentFromTeamByIdCommand.java:
```diff
@@ -0,0 +1,91 @@
+
+        personToDelete = model.searchPersonByPredicate(predicate);
+        if (personToDelete == null) {
+            throw new CommandException(String.format(PersonMessages.MESSAGE_PERSON_EMAIL_NOT_FOUND, studentId));
```
Good catch! Gonna fix that now

whelan-low committed:
Fix error message for delete by id command

whelan-low committed:
Merge branch 'master' into delete-student-from-team

whelan-low committed:
Merge branch 'master' into delete-student-from-team

whelan-low committed:
Fix merge issues

whelan-low committed:
Put delete student from team success message back into DeleteStudentFromTeamCommand class

Zack-Tay commented on src/test/java/seedu/address/logic/commands/DeleteStudentFromTeamCommandTest.java:
```diff
@@ -0,0 +1,217 @@
+        DeleteStudentFromTeamByIndexCommand allocateStudentToTeamByIndexCommand =
+                new DeleteStudentFromTeamByIndexCommand(Index.fromZeroBased(0),
+                        newModule, tutorialClass, newTeam);
+        assertEquals(deleteOtherStudentFromTeamByEmailCommand.toString(),
+                deleteOtherStudentFromTeamByEmailCommand.toString());
+        assertEquals(deleteStudentFromTeamByIdCommand.toString(),
+                deleteStudentFromTeamByIdCommand.toString());
+        assertEquals(allocateStudentToTeamByIndexCommand.toString(),
+                allocateStudentToTeamByIndexCommand.toString());
```
are these testcases meaningful? I think you are just comparing 2 same objects which should always be true anyways.

whelan-low replied:
I'm just following the test cases used in AB3. I also felt it wasnt necessary but I'm not sure if I should delete

Zack-Tay commented on src/test/java/seedu/address/logic/commands/DeleteStudentFromTeamCommandTest.java:
```diff
@@ -0,0 +1,217 @@
+
+
+        DeleteStudentFromTeamByIndexCommand deleteOtherStudentFromTeamByIndexCommand =
+                new DeleteStudentFromTeamByIndexCommand(Index.fromZeroBased(1),
```
I think the use of magic numbers should be reduced.

Zack-Tay reviewed (commented):
I like that the messages have been refactored into their respective folders to make references more noticeable and easy to find. However, I notice a few nits in your PR, do look through them! Thanks for the effort and hardwork!

whelan-low commented on src/test/java/seedu/address/logic/commands/DeleteStudentFromTeamCommandTest.java:
```diff
@@ -0,0 +1,217 @@
+    @Test
+    public void execute_deleteStudentFromClassById_success() {
+        newTeam.addStudent(validPerson);
+        assertEquals(1, newTeam.getStudents().size());
```
Good catch. I will alter these accordingly

whelan-low commented on src/main/java/seedu/address/logic/commands/deletestudentfromteamcommands/DeleteStudentFromTeamByEmailCommand.java:
```diff
@@ -0,0 +1,90 @@
+        if (personToDelete == null) {
+            throw new CommandException(String.format(PersonMessages.MESSAGE_PERSON_EMAIL_NOT_FOUND, email));
+        }
+        if (team == null) {
+            throw new CommandException(String.format(TeamMessages.MESSAGE_TEAM_DOES_NOT_EXIST, tutorialTeam,
+                    tutorialClass));
+        }
```
Alright, will refactor. Thanks

whelan-low commented on src/main/java/seedu/address/logic/commands/deletestudentfromteamcommands/DeleteStudentFromTeamByIndexCommand.java:
```diff
@@ -0,0 +1,90 @@
+
+        Person personToDelete;
+
+        try {
+            personToDelete = model.getFilteredPersonList().get(targetIndex.getZeroBased());
+        } catch (IndexOutOfBoundsException e) {
+            throw new CommandException(
+                    String.format(PersonMessages.MESSAGE_PERSON_INDEX_NOT_FOUND, targetIndex.getOneBased()));
```
Ah, I thought the index was with reference to the address book list. Alright, will fix this. Thanks!

whelan-low commented on src/main/java/seedu/address/logic/commands/deletestudentfromteamcommands/DeleteStudentFromTeamCommand.java:
```diff
@@ -0,0 +1,74 @@
+        this.tutorialTeam = tutorialTeam;
+    }
+
+    protected ModuleTutorialPair getModuleAndTutorialClass(Model model) throws CommandException {
+        requireNonNull(model);
+        ModuleCode module = getModule();
+        TutorialClass tutorialClass = getTutorialClass();
+        ModuleCode existingModule = model.findModuleFromList(module);
+        TutorialClass existingTutorialClass = model.findTutorialClassFromList(tutorialClass, existingModule);
+        if (existingModule == null) {
+            throw new CommandException(String.format(ModuleMessages.MESSAGE_MODULE_NOT_FOUND, module));
+        }
+        if (existingTutorialClass == null) {
+            throw new CommandException(
+                    String.format(ModuleMessages.MESSAGE_TUTORIAL_DOES_NOT_BELONG_TO_MODULE, tutorialClass, module));
```
Yup, it was found in addStudentToClass. I will refactor it into moduleTutorialPair

whelan-low committed:
Move getModuleAndTutorialPair into ModuleTutorialPair class, fixed magic number issues and implementation errors

whelan-low commented:
In my latest commit, I have fixed the magic numbers issue, and repaired some implementation errors regarding index, Also refactored getModuleTutorialPair into ModuleTutorialPair class.

whelan-low commented:
Sorry for making such a large PR

Jajared reviewed (approved):
LGTM!
